### PR TITLE
Restructure build tools java packages

### PR DESCRIPTION
**Pull request by breskeby**
Related to #71593 we move all build logic that is for elasticsearch build only into
the org.elasticsearch.gradle.internal* packages

This makes it clearer if build logic is considered to be used by external projects
Ultimately we want to only expose TestCluster and PluginBuildPlugin logic to third party plugin authors. 
Related to this we'll likely have the DistributionDownloadPlugin that TestCluster plugin is depending on public too. 
There are a couple of related domain and util classes that are exposed to.

This is a very first step towards splitting up public plugins from the internal elasticsearch build logic.
Next steps are to remove dependencies to internal classes from our public shared code and ultimately move this code
into a dedicated project.

### CODE REVIEW

File: buildSrc/src/integTest/groovy/org/elasticsearch/gradle/DistributionDownloadPluginFuncTest.groovy [hunk 1]
```diff
@@ -32,7 +33,7 @@ class DistributionDownloadPluginFuncTest extends AbstractGradleFuncTest {
 
         where:
         version                              | platform                                   | distType
-        VersionProperties.getElasticsearch() | ElasticsearchDistribution.Platform.LINUX   | "current"
+        VersionProperties.getElasticsearch() | ElasticsearchDistribution.Platform.LINUX | "current"
```
Comment (mark-vieira) [inline buildSrc/src/integTest/groovy/org/elasticsearch/gradle/DistributionDownloadPluginFuncTest.groovy:36]: Is this intentional?

**Review by mark-vieira — COMMENTED** [review]
I'm wondering if we really need to rename the plugin ids or even as many of these packages. "Internal" in this context means stuff that gets put in `build-tools` jar but external users shouldn't use. Going forward though, none of this stuff will be in that jar, as we won't just be naively publishing all of `buildSrc`. In that scenario "internal" should just mean stuff that ES developers should reach into directly and that certainly doesn't include our own custom plugins.

Is this move just temporary to help delineate that line, then we can back off once we have a separate "public" build-tools artifact?

**Comment by breskeby** [comment]
Once we've clearly separated out the public build tools we can probably back off the internal part again. But this move here helps getting us (me) there :) 
Likely we can have sth like org.elasticsearch.gradle.buildtools for the public stuff.

**Comment by breskeby** [comment]
Renaming the IDs can provably avoided. So far we mostly differentiate if there's an internal and an external version of a plugin (e.g. DistributionDownloadPlugin vs. InternalDistributionDownloadPlugin)

File: buildSrc/src/integTest/groovy/org/elasticsearch/gradle/DistributionDownloadPluginFuncTest.groovy (hunk 1 above)
Comment (breskeby) [reply buildSrc/src/integTest/groovy/org/elasticsearch/gradle/DistributionDownloadPluginFuncTest.groovy]: fixed.

**Comment by mark-vieira** [comment]
> Likely we can have sth like org.elasticsearch.gradle.buildtools for the public stuff.

That sounds good. Having a package specific to the public stuff makes sense.

> Renaming the IDs can provably avoided. So far we mostly differentiate if there's an internal and an external version of a plugin (e.g. DistributionDownloadPlugin vs. InternalDistributionDownloadPlugin)

:+1:

**Pull request merged**